index.js

```diff
@@ -185,8 +185,16 @@ HtmlWebpackPlugin.prototype.evaluateCompilationResult = function(compilation, co
   if(!compilationResult) {
     return Promise.reject('The child compilation didn\'t provide a result');
   }
-  // Strip the leading 'var '
-  var source = compilationResult.source().substr(3);
+  var source = compilationResult.source();
+  // Strip the leading 'var ' if present.
+  // If webpack.BannerPlugin is used, `source` starts with given comment.
+  if (_.startsWith(source, 'var')) {
```

Comment:
I like your idea and comment - however I would not add the if statement.
Just add `source = source.replace('var result =', '');` for both cases